### ISSUE TITLE
Using it within println hangs compilation

### ISSUE THREAD

**Issue by Fiedzia** [dtolnay/indoc#21]
println!(indoc!("hello world"));

hangs compilation

**Comment by dtolnay** [comment]
Thanks! Check out #17. As a workaround you can use `println!("{}", indoc!(...))` or use the `"unstable"` feature of indoc on a nightly compiler:

```toml
[dependencies]
indoc = { version = "0.2", features = ["unstable"] }
```

```rust
#![feature(proc_macro, proc_macro_non_items)]

extern crate indoc;
use indoc::indoc;

fn main() {
    println!(indoc!(""));
}
```

**Comment by dtolnay** [comment]
The hang is a compiler bug: https://github.com/rust-lang/rust/issues/44692.

**Comment by MightyPork** [comment]
it doesn't hang for me, but indoc in format macros is not allowed because the (stable) compiler thinks it is not a string literal. Really unfortunate, as that's the use case I had for it :/

**Comment by dtolnay** [comment]
It seems the hang was fixed in rustc 1.38.0.

```console
error: format argument must be a string literal
 --> src/main.rs:4:14
  |
4 |     println!(indoc!(""));
  |              ^^^^^^^^^^
  |
  = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
help: you might be missing a string literal to format with
  |
4 |     println!("{}", indoc!(""));
  |              ^^^^^
```

**Comment by dtolnay** [comment]
@MightyPork  I filed #36 to consider adding println-like macros with indented strings.
